src/reducers/deleteThought.ts

```diff
@@ -229,7 +233,7 @@ const deleteThought = (state: State, { context, thoughtId, orphaned }: Payload)
     state => ({
       ...state,
       contextViews: contextViewsNew,
-      cursor: cursorNew,
+      cursor: prevThought ? ([prevThought.id] as Path) : cursorNew,
```

Comment:
This isn't a valid way to construct a Path, as a Path must start at the ROOT. You probably want something like `[...parentOf(state.cursor), prevThought.id]`.

Reply:
Also, instead of adding the condition here and ignoring `cursorNew`, check the condition when creating `cursorNew` and set the correct value there. 

This line should just be a simple assignment.

Reply:
To answer your question, in `deleteThoughts.ts` reducer `path` is calculated via `rankThoughtsFirstMatch` selector function which is returning the `deleted thought path` for non-empty thought while it's `prev thought path` for an empty thought. So now I am trying to figure out which is the expected behavior. What value should `path` variable in `deleteThoughts` must hold on these two scenarios?

Reply:
My intention with the new logic was to actually replace the existing logic and the way cursor is being calculated. I thought the better way would to be find out the prev thought/sibling and then use it's id as path for next cursor update.

Reply:
> To answer your question, in deleteThoughts.ts reducer path is calculated via rankThoughtsFirstMatch selector function which is returning the deleted thought path for non-empty thought while it's prev thought path for an empty thought.

Aha! That's definitely the problem. rankThoughtsFirstMatch is a heuristic that can't distinguish siblings. 

Luckily, it's obsolete with the new id-based architecture. Try to find a way to get the Path without having to convert from Context using `rankThoughtsFirstMatch`. Usually you can either build up the Path from the cursor, or you can use one of the selector functions. I can look more closely at this if you need help identifying the correct selector to use. 

> My intention with the new logic was to actually replace the existing logic and the way cursor is being calculated. I thought the better way would to be find out the prev thought/sibling and then use it's id as path for next cursor update.

Fair enough. My comments still hold. You need to solve the problem upstream, and the Path you created is not valid. All Paths must start at the ROOT thought.

Reply:
I have just pushed some changes, however I don't think that it still fully justifies the approach that you mentioned just now. I will have a look once again and build a optimal solution.

Reply:
> Luckily, `rankThoughtsFirstMatch` is largely obsolete with the new id-based architecture. Try to find a way to get the `Path` without having to convert from a `Context` using `rankThoughtsFirstMatch`. Usually you can either build up the `Path` from the cursor, or you can use one of the selector functions. I can look more closely at this if you need help identifying the correct selector to use.

I can see there are number of ways to construct a `path`. I am not sure on what selector function to use for this specific purpose.